pkg/api/v2/types.go

```diff
@@ -0,0 +1,313 @@
+	PodMinNumber       int32                  `yaml:"podMinNumber"`
+	PodMaxNumber       int32                  `yaml:"podMaxNumber"`
+	CellsPerPod        PodGroupMemberCellSpec `yaml:"cellsPerPod"`
+	ContainsCurrentPod bool                   `yaml:"containsCurrentPod"`
+}
+
+// PodGroupMemberCellSpec represents cell spec for each pod in pod group.
+type PodGroupMemberCellSpec struct {
+	CellType   api.CellType `yaml:"cellType"`
+	CellNumber int32        `yaml:"cellNumber"`
+}
+
+type PodGroupStatus struct {
+	VC                   api.VirtualClusterName                `json:"vc"`
+	Priority             int32                                 `json:"priority"`
+	State                string                                `json:"state"`
+	PhysicalPlacement    map[string][]int32                    `json:"physicalPlacement,omitempty"` // node -> leaf cell indices
+	VirtualPlacement     map[api.CellAddress][]api.CellAddress `json:"virtualPlacement,omitempty"`  // preassigned cell -> leaf cells
+	AllocatedPods        []types.UID                           `json:"allocatedPods,omitempty"`
+	PreemptingPods       []types.UID                           `json:"preemptingPods,omitempty"`
+	LazyPreemptionStatus *api.LazyPreemptionStatus             `json:"lazyPreemptionStatus,omitempty"`
+}
+
+type podGroupSpecIterator struct {
+	pods   []*PodGroupMemberSpec
+	index  int
+	length int
+}
+
+// Next returns the next item in iteration.
+func (i *podGroupSpecIterator) Next() *PodGroupMemberSpec {
+	i.index++
+	return i.pods[i.index-1]
+}
+
+// HasNext return true if iteration not finishes.
+func (i *podGroupSpecIterator) HasNext() bool {
+	return i.index < i.length
+}
+
+// Iterator returns a stateful iterator for PodGroupSpec
+func (podRootGroup *PodGroupSpec) Iterator() *podGroupSpecIterator {
+	pods := []*PodGroupMemberSpec{}
+	queue := []*PodGroupSpec{podRootGroup}
+	for len(queue) > 0 {
+		newQueue := []*PodGroupSpec{}
+		for _, podGroup := range queue {
+			for podIndex := range podGroup.Pods {
+				pods = append(pods, &podGroup.Pods[podIndex])
+			}
+			newQueue = append(newQueue, podGroup.ChildGroups...)
+		}
+		queue = newQueue
+	}
+	return &podGroupSpecIterator{pods, 0, len(pods)}
+}
+
+// SetCellType sets cell type for all pods in pod group.
+func (podRootGroup *PodGroupSpec) SetCellType(cellType string) {
+	for iter := podRootGroup.Iterator(); iter.HasNext(); {
+		iter.Next().CellsPerPod.CellType = api.CellType(cellType)
+	}
+}
+
+// GetCurrentPod returns level traverse index and current pod in pod group.
+func (obj *PodSchedulingSpec) GetCurrentPod() (int32, PodGroupMemberSpec) {
+	index := int32(0)
+	queue := []*PodGroupSpec{obj.PodRootGroup}
+	for len(queue) > 0 {
+		newQueue := []*PodGroupSpec{}
+		for _, podGroup := range queue {
+			for _, pod := range podGroup.Pods {
+				if pod.ContainsCurrentPod == true {
+					return index, pod
+				}
+			}
+			index++
+			newQueue = append(newQueue, podGroup.ChildGroups...)
+		}
+		queue = newQueue
+	}
```

Comment:
Why not use Iterator to find?

Reply:
needs a level traverse index (podGroup index) here, Iterator will traverse in a flattened pod slice (can only get pod index).